cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -64,6 +66,8 @@ public MRJobClient(CConfiguration cConf, Configuration hConf) {
    * @throws NotFoundException if a Job with the given runId is not found.
    */
   public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundException {
+    Preconditions.checkArgument(ProgramType.MAPREDUCE.equals(runId.getProgram().getType()));
```

Comment:
The runId of the workflow wouldn't be found from the JobClient.
Would have to use the MR's runId (even if its in a workflow).